packages/flutter/lib/src/rendering/object.dart

```diff
@@ -82,6 +82,22 @@ class PaintingContext extends ClipContext {
   /// The [estimatedBounds] rectangle is in the [canvas] coordinate system.
   final Rect estimatedBounds;
 
+  /// Paint [child] with this painting context, then convert to a raster and detach all
+  /// children from this layer.
+  static ui.Image paintAndDetachToGpuImage(OffsetLayer layer, {
```

Comment:
Why is this needed?

If you have an offset layer, why not just call `layer.toGpuImage`? If it's not the offset layer that is already composited for this subtree, why are we asking you for it at all?

Reply:
Alternative question: should this really just be call to `PaintingContext.repaintCompositedChild`, which knows how track whether this layer has already been composited or not?

Reply:
Its possible I could make this work by calling super.paint with a locally constructed PaintingContext as well, but this seems useful because there are a few gotchas.  Its important that the detach is added as well, otherwise you may draw things twice.  

Calling `PaintingContext.repaintCompositedChild` would be a re-entrant call to the same render object. If we wanted to use that we'd need to have the animated raster RO create two layers.